contracts/cw20-atomic-swap/src/contract.rs

```diff
@@ -31,22 +37,48 @@ pub fn handle<S: Storage, A: Api, Q: Querier>(
     msg: HandleMsg,
 ) -> StdResult<HandleResponse> {
     match msg {
-        HandleMsg::Create(msg) => try_create(deps, env, msg),
+        HandleMsg::Create(msg) => try_create(
+            deps,
+            env.clone(),
```

Comment:
Why clone? We never use it elsewhere

Reply:
It's needed because of the passing by value, just below (`Balance::Native(env.message.sent_funds)`). Structs have this behaviour in Rust by default... will try to refactor into an assignment and a construction.

Reply:
Coin contains strings, and cannot be Copy, only Clone.
I missed the other usage below... my bad. It was fine as it was, I just needed :coffee:

Reply:
No problem. I'll just close the issue I wrote regarding Copy coins then.